### PR TITLE
Merging bdd-style tests back to master

### PR DESCRIPTION
**Pull request by godrose**

### CODE REVIEW

File: src/LogoFX.Client.Mvvm.Model.Tests/Objects/EditableModelWithReadOnlyField.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        {
+            get => _status;
+            set
+            {
+                if (value == _status)
+                {
+                    return;
+                }
+
+                MakeDirty();
```
Comment (dmitrybublik) [inline src/LogoFX.Client.Mvvm.Model.Tests/Objects/EditableModelWithReadOnlyField.cs]: Why not use SetProperty?

File: src/LogoFX.Client.Mvvm.Model.Tests/Objects/EditableModelWithValidation.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            get { return _title; }
+            set
+            {
+                if (_title == value)
+                {
+                    return;
+                }
+
+                MakeDirty();
+                _title = value;
```
Comment (dmitrybublik) [inline src/LogoFX.Client.Mvvm.Model.Tests/Objects/EditableModelWithValidation.cs]: SetProperty

File: src/LogoFX.Client.Mvvm.Model.Tests/Objects/SelfEditableModel.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+            set
+            {
+                if (_value == value)
+                {
+                    return;
+                }
+
+                MakeDirty();
+                _value = value;
```
Comment (dmitrybublik) [inline src/LogoFX.Client.Mvvm.Model.Tests/Objects/SelfEditableModel.cs]: SetProperty

Comment (godrose) [reply src/LogoFX.Client.Mvvm.Model.Tests/Objects/SelfEditableModel.cs]: @dmitrybublik Definitely. Thanks.

File: src/LogoFX.Client.Mvvm.Model.Tests/Objects/EditableModelWithValidation.cs (hunk 2 above)
Comment (godrose) [reply src/LogoFX.Client.Mvvm.Model.Tests/Objects/EditableModelWithValidation.cs]: @dmitrybublik Definitely. Thanks.

File: src/LogoFX.Client.Mvvm.Model.Tests/Objects/EditableModelWithReadOnlyField.cs (hunk 1 above)
Comment (godrose) [reply src/LogoFX.Client.Mvvm.Model.Tests/Objects/EditableModelWithReadOnlyField.cs]: @dmitrybublik Definitely. Thanks.

File: src/LogoFX.Client.Mvvm.Model.Tests/Specs/Editing Lifecycle.feature [hunk 4]
```diff
@@ -0,0 +1,120 @@
+﻿Feature: Editing Lifecycle
+	In order to develop rich functionl apps faster
```
Comment (dmitrybublik) [inline src/LogoFX.Client.Mvvm.Model.Tests/Specs/Editing Lifecycle.feature]: Mistake 'functionl'

**Pull request merged**